client/src/app/audiologist-navigation/audiologist-navigation.component.ts

```diff
@@ -64,22 +64,25 @@ export class AudiologistNavigationComponent implements OnInit {
   }
 
   public clearData() {
-    // clear all patient data in memory
-    let sessionKeys: string[] = [
-      'patient-id',
-      'tests-data',
-      'tfi-dataRecord',
-      'ths-dataRecord',
-      'ths-history',
-      'ts-dataRecord',
-      'ts-history',
-      'appt'
-    ];
-    sessionKeys.forEach((value) => {
-      Utilities.removeItemFromSessionStorage(value);
-    });
-    this.patientID = null;
-    this.state.determineState(false);
+    if (confirm('Are you sure you want to discard all data?')) {
```

Comment:
This should be handled differently if you logout:

Currently if you logout, it asks 'Are you sure you want to discard all data?'. Regardless of what you choose, you logout either way. However if you choose OK, it also clears your data. Being able to logout without clearing data is a nice feature if we can communicate it: you can login as someone else and still have the patient's data, as long as you do not start another survey.

This also triggers when you submit survey. This message should not appear when you submit survey; instead it should always discard.

Reply:
I'll make another ticket, because this ticket was only meant to make a pop up for the discard button.

Reply:
Ohhhhhhh, actually no. @MishalAlajmi-psu, Griffin is right , this confirmation shouldn't be in the `clearData()` function. (`Cleardata()` is also called by the `logOut()` function. Discard should be mapped to its *own function*).